### ISSUE TITLE
Some Unices don't append ".0.0" at the end of shared libraries, but just ".0"

### ISSUE THREAD

**Issue by kusalananda** [beagle-dev/beagle-lib#132]
I finally managed to get the CPU (etc.) plugin to load in MrBayes on OpenBSD.  I did so by modifying the expected shared library filename in `libhmsbeagle/plugin/UnixSharedLibrary.h` to only include one `.0` at the end of the shared library filename.

The shared libraries on this system end up being named as `libhmsbeagle-cpu.so.31.0` (note the single `.0`), so I added the following code the the above-mentioned header (this is `git diff` output):

```
diff --git a/libhmsbeagle/plugin/UnixSharedLibrary.h b/libhmsbeagle/plugin/UnixSharedLibrary.h
index 1607785..f9b4d15 100644
--- a/libhmsbeagle/plugin/UnixSharedLibrary.h
+++ b/libhmsbeagle/plugin/UnixSharedLibrary.h
@@ -44,6 +44,10 @@ UnixSharedLibrary::UnixSharedLibrary(const char* name)
     libname += ".";
     libname += PLUGIN_VERSION;
     libname += ".so";
+#elif defined(__OpenBSD__)
+    libname += ".so.";
+    libname += PLUGIN_VERSION;
+    libname += ".0";
 #else
     libname += ".so.";
     libname += PLUGIN_VERSION;
```

I don't know if this affects other Unices too.

**Comment by kusalananda** [comment]
This obviously also affects `LibtoolSharedLibrary.h` is `libtool` is used.

```
diff --git a/libhmsbeagle/plugin/LibtoolSharedLibrary.h b/libhmsbeagle/plugin/LibtoolSharedLibrary.h
index 7e03bd0..0318052 100644
--- a/libhmsbeagle/plugin/LibtoolSharedLibrary.h
+++ b/libhmsbeagle/plugin/LibtoolSharedLibrary.h
@@ -49,6 +49,10 @@ UnixSharedLibrary::UnixSharedLibrary(const char* name)
     libname += ".";
     libname += PLUGIN_VERSION;
     libname += ".so";
+#elif defined(__OpenBSD__)
+    libname += ".so.";
+    libname += PLUGIN_VERSION;
+    libname += ".0";
 #else
     libname += ".so.";
     libname += PLUGIN_VERSION;
```

**Comment by kusalananda** [comment]
Just to say that this make using Beagle no NetBSD possible as well:
```
diff --git a/libhmsbeagle/plugin/LibtoolSharedLibrary.h b/libhmsbeagle/plugin/LibtoolSharedLibrary.h
index 7e03bd0..bf7f77a 100644
--- a/libhmsbeagle/plugin/LibtoolSharedLibrary.h
+++ b/libhmsbeagle/plugin/LibtoolSharedLibrary.h
@@ -49,6 +49,10 @@ UnixSharedLibrary::UnixSharedLibrary(const char* name)
     libname += ".";
     libname += PLUGIN_VERSION;
     libname += ".so";
+#elif defined(__OpenBSD__) || defined(__NetBSD__)
+    libname += ".so.";
+    libname += PLUGIN_VERSION;
+    libname += ".0";
 #else
     libname += ".so.";
     libname += PLUGIN_VERSION;
diff --git a/libhmsbeagle/plugin/UnixSharedLibrary.h b/libhmsbeagle/plugin/UnixSharedLibrary.h
index 1607785..ffff347 100644
--- a/libhmsbeagle/plugin/UnixSharedLibrary.h
+++ b/libhmsbeagle/plugin/UnixSharedLibrary.h
@@ -44,6 +44,10 @@ UnixSharedLibrary::UnixSharedLibrary(const char* name)
     libname += ".";
     libname += PLUGIN_VERSION;
     libname += ".so";
+#elif defined(__OpenBSD__) || defined(__NetBSD__)
+    libname += ".so.";
+    libname += PLUGIN_VERSION;
+    libname += ".0";
 #else
     libname += ".so.";
     libname += PLUGIN_VERSION;
```

**Comment by msuchard** [comment]
Thanks!  Merged into a05c94fd5746a29722633f380172a20583787982 in preparation for release of BEAGLE 3.2
